hlog/internal/mutil/body.go

```diff
@@ -0,0 +1,30 @@
+package mutil
+
+import "io"
+
+type byteCountReadCloser struct {
+	rc   io.ReadCloser
+	read int64
+}
+
+var _ io.ReadCloser = (*byteCountReadCloser)(nil)
+
+func NewByteCountReadCloser(body io.ReadCloser) *byteCountReadCloser {
```

Comment:
In practice, http.body struct also implements `WriteTo`. So it might be reasonable to add counter here as well.